xds/src/main/java/io/grpc/xds/ClientXdsClient.java

```diff
@@ -583,15 +574,20 @@ private static FilterChainMatch parseFilterChainMatch(
     if (parseHttpFilter) {
       filterConfigs = new ArrayList<>();
       Set<String> names = new HashSet<>();
-      for (io.envoyproxy.envoy.extensions.filters.network.http_connection_manager.v3.HttpFilter
-               httpFilter : proto.getHttpFiltersList()) {
+      NamedFilterConfig lastFilter = new NamedFilterConfig(null, null);
```

Comment:
The lastFilter check outside of the loop was intended to cover the empty list case. Because previous empty filter chains would end up a lame_filter later on, but we removed that part here.
But I think the way you described here is better - functionally the same but error message more friendly. Updated.